src/main/resources/templates/db_scripts/oaipmh/createTriggersAndFunctionsForCompleteUpdatedDate.sql

```diff
@@ -0,0 +1,111 @@
+ DROP TRIGGER IF EXISTS completeUpdatedDate_holdings_record_inserted_updated
+ ON ${myuniversity}_${mymodule}.holdings_record;
+ DROP TRIGGER IF EXISTS completeUpdatedDate_holdings_record_deleted
+ ON ${myuniversity}_${mymodule}.holdings_record;
+ DROP TRIGGER IF EXISTS completeUpdatedDate_item_inserted_updated
+ ON ${myuniversity}_${mymodule}.item;
+ DROP TRIGGER IF EXISTS completeUpdatedDate_item_deleted
+ ON ${myuniversity}_${mymodule}.item;
+
+ DROP FUNCTION IF EXISTS ${myuniversity}_${mymodule}.completeUpdatedDate_for_insert_update_holding;
+ DROP FUNCTION IF EXISTS ${myuniversity}_${mymodule}.completeUpdatedDate_for_delete_holding;
+ DROP FUNCTION IF EXISTS ${myuniversity}_${mymodule}.completeUpdatedDate_for_insert_update_item;
+ DROP FUNCTION IF EXISTS ${myuniversity}_${mymodule}.completeUpdatedDate_for_delete_item;
+
+ CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.completeUpdatedDate_for_insert_update_holding()
+     RETURNS trigger
+     LANGUAGE 'plpgsql'
+     COST 100
+     VOLATILE NOT LEAKPROOF
+ AS $BODY$
+ BEGIN
+      UPDATE ${myuniversity}_${mymodule}.instance inst SET completeUpdatedDate =
+          (NEW.jsonb -> 'metadata' ->> 'updatedDate')::timestamp with time zone
+      WHERE inst.id = (NEW.jsonb ->> 'instanceId')::uuid;
+   RETURN NEW;
+ END;
+ $BODY$;
+
+ CREATE TRIGGER completeUpdatedDate_holdings_record_inserted_updated
+     BEFORE UPDATE OR INSERT
+     ON ${myuniversity}_${mymodule}.holdings_record
+     FOR EACH ROW
+     EXECUTE FUNCTION ${myuniversity}_${mymodule}.completeUpdatedDate_for_insert_update_holding();
+
+
+ CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.completeUpdatedDate_for_delete_holding()
+     RETURNS trigger
+     LANGUAGE 'plpgsql'
+     COST 100
+     VOLATILE NOT LEAKPROOF
+ AS $BODY$
+ DECLARE
+    createdDate text;
+ BEGIN
+      SELECT (jsonb ->> 'createdDate')::text INTO createdDate
+      FROM ${myuniversity}_${mymodule}.audit_holdings_record
+      WHERE (jsonb -> 'record' ->> 'id')::uuid = OLD.id;
+
+      UPDATE ${myuniversity}_${mymodule}.instance inst SET completeUpdatedDate =
```

Comment:
You can rewrite this the following way without querying **audit_holdings_record** table. 
 
UPDATE ${myuniversity}_${mymodule}.instance inst SET completeUpdatedDate = now()
WHERE inst.id = OLD.instanceid;

Reply:
Done